python/hail/expr/tests.py

```diff
@@ -224,5 +224,496 @@ def test_str_ops(self):
         self.assertFalse(eval_expr(s5.to_boolean()))
         self.assertFalse(eval_expr(s6.to_boolean()))
 
-
+    def check_expr(self, expr, expected, expected_type):
+        self.assertEqual(expected_type, expr.dtype)
+        self.assertEqual((expected, expected_type), eval_expr_typed(expr))
+
+    def test_arithmetic(self):
+        a_int32 = functions.capture([2, 4, 8, 16])
+        a_int64 = a_int32.map(lambda x: x.to_int64())
+        a_float32 = a_int32.map(lambda x: x.to_float32())
+        a_float64 = a_int32.map(lambda x: x.to_float64())
+        int64_4 = functions.capture(4).to_int64()
+        int64_4s = functions.capture([4, 4, 4, 4]).map(lambda x: x.to_int64())
+        float32_4 = functions.capture(4).to_float32()
+        float32_4s = functions.capture([4, 4, 4, 4]).map(lambda x: x.to_float32())
+        float64_4 = functions.capture(4).to_float64()
+        float64_4s = functions.capture([4, 4, 4, 4]).map(lambda x: x.to_float64())
+        int64_3 = functions.capture(3).to_int64()
+        int64_3s = functions.capture([3, 3, 3, 3]).map(lambda x: x.to_int64())
+        float32_3 = functions.capture(3).to_float32()
+        float32_3s = functions.capture([3, 3, 3, 3]).map(lambda x: x.to_float32())
+        float64_3 = functions.capture(3).to_float64()
+        float64_3s = functions.capture([3, 3, 3, 3]).map(lambda x: x.to_float64())
+
+        ##########
+        # Division
```

Comment:
Our code base pretty frequently violates this, but it's good practice to use lots of small test functions because we get finer-grained information about what went wrong and where. If we desire it, converting to python is an opportunity to change our practice.

I'm also not sure if Python does this, but most test libraries terminate as soon as one assertion in a test method fails. I generally prefer smaller test methods so that I get a wider sense of which test are failing (and thus which properties of the code I've broken).